hoodie-common/src/main/java/com/uber/hoodie/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -53,62 +49,81 @@
+  // Total log blocks read
+  private final AtomicLong totalLogBlocks = new AtomicLong(0);
   // Latest valid instant time
-  private String latestInstantTime;
-  private HoodieTableMetaClient hoodieTableMetaClient;
+  private final String latestInstantTime;
+  private final HoodieTableMetaClient hoodieTableMetaClient;
   // Merge strategy to use when combining records from log
-  private String payloadClassFQN;
+  private final String payloadClassFQN;
+  // Log File Paths
+  private final List<String> logFilePaths;
+  // Read Lazily flag
+  private final boolean readBlocksLazily;
+  // Reverse reader - Not implemented yet (NA -> Why do we need ?)
+  // but present here for plumbing for future implementation
+  private final boolean reverseReader;
+  // FileSystem
+  private final FileSystem fs;
   // Store the last instant log blocks (needed to implement rollback)
   private Deque<HoodieLogBlock> currentInstantLogBlocks = new ArrayDeque<>();
+  // Progress
+  private float progress = 0.0f;
 
-  public HoodieCompactedLogRecordScanner(FileSystem fs, String basePath, List<String> logFilePaths,
-      Schema readerSchema, String latestInstantTime, Long maxMemorySizeInBytes,
+  public AbstractHoodieLogRecordScanner(FileSystem fs, String basePath, List<String> logFilePaths,
+      Schema readerSchema, String latestInstantTime,
       boolean readBlocksLazily, boolean reverseReader) {
     this.readerSchema = readerSchema;
     this.latestInstantTime = latestInstantTime;
     this.hoodieTableMetaClient = new HoodieTableMetaClient(fs.getConf(), basePath);
     // load class from the payload fully qualified class name
     this.payloadClassFQN = this.hoodieTableMetaClient.getTableConfig().getPayloadClass();
+    this.logFilePaths = logFilePaths;
+    this.readBlocksLazily = readBlocksLazily;
+    this.reverseReader = reverseReader;
+    this.fs = fs;
+  }
 
+  /**
+   * Scan Log files
+   */
+  public void scan() {
     try {
-      // Store merged records for all versions for this log file, set the in-memory footprint to maxInMemoryMapSize
-      this.records = new ExternalSpillableMap<>(maxMemorySizeInBytes, Optional.empty(),
-          new StringConverter(), new HoodieRecordConverter(readerSchema, payloadClassFQN));
       // iterate over the paths
       HoodieLogFormatReader logFormatReaderWrapper =
           new HoodieLogFormatReader(fs,
               logFilePaths.stream().map(logFile -> new HoodieLogFile(new Path(logFile)))
                   .collect(Collectors.toList()), readerSchema, readBlocksLazily, reverseReader);
+      Set<HoodieLogFile> seenLogFiles = new HashSet<>();
       while (logFormatReaderWrapper.hasNext()) {
         HoodieLogFile logFile = logFormatReaderWrapper.getLogFile();
         log.info("Scanning log file " + logFile);
-        totalLogFiles.incrementAndGet();
+        seenLogFiles.add(logFile);
+        totalLogFiles.set(seenLogFiles.size());
+        progress = seenLogFiles.size() - 1 / logFilePaths.size();
```

Comment:
This is just the progress of reading the log blocks but doesn't account merging/processing them, what is the intention of exposing progress ?